onnxscript/optimizer/_constant_folding.py

```diff
@@ -235,11 +259,27 @@ def decorator(function: PartialEvaluatorFunction) -> PartialEvaluatorFunction:
 register = registry.register
 
 
-def _get_numpy_value(val: ir.Value | None) -> np.ndarray | None:
+def _same_shape(shape1: ir.Shape, shape2: ir.Shape) -> bool:
+    # Comparison of shapes as tuples works except if any dimension is None
+    # (which represents an unknown dimension value). Thus, two shapes such
+    # as (Batch, 1024) and (Batch, 1024) are considered equal, but (None, 1024)
+    # and (None, 1024) are not considered equal.
+    if any(isinstance(dim, ir.SymbolicDim) and dim.value is None for dim in shape1):
+        return False
+    return shape1.dims == shape2.dims
+
+
+def _get_numpy_value(
+    val: ir.Value | None, dtype: ir.DataType | None = None, size_limit: int | None = None
```

Comment:
Seems helpful to document the behavior of `dtype`; intuitively it can also mean casting but here it's used as a condition

Reply:
Done. The dtype is not critical, main goal was to add size limit to avoid triggering conversion of large tensors into numpy array, like when optimizing a Gather.  But a dtype filter can also help avoid some unnecessary conversions.